src/TesApi.Web/BatchPoolService.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace TesApi.Web
+{
+    /// <summary>
+    /// A background service that montitors CloudPools in the batch system, orchestrates their lifecycle, and updates their state.
+    /// This should only be used as a system-wide singleton service.  This class does not support scale-out on multiple machines,
+    /// nor does it implement a leasing mechanism.  In the future, consider using the Lease Blob operation.
+    /// </summary>
+    public class BatchPoolService : BackgroundService
+    {
+        private readonly IBatchScheduler _batchScheduler;
+        private readonly ILogger _logger;
+        private readonly bool _isDisabled;
+
+        /// <summary>
+        /// Interval between each call to <see cref="IBatchPool.ServicePoolAsync(CancellationToken)"/>.
+        /// </summary>
+        public static readonly TimeSpan RunInterval = TimeSpan.FromSeconds(30);
```

Comment:
Any thoughts on minimum value here, or do you think 30 is optimal?  Presumably, the lower the number, the greater the chance of API throttling

Reply:
The code handles API throttling via retries, whether in `CachingWithRetriesAzureProxy` or by deferring until the next run of the service, although in the grand scheme of things retries snowball the throttling problems.

Reply:
A good future test would be to reduce this run interval to 1 second and see what, if anything, breaks